test/src/unittests/machinelearning/test_tensorflowpredicteffnetdiscogs.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2006-2021  Music Technology Group - Universitat Pompeu Fabra
+#
+# This file is part of Essentia
+#
+# Essentia is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from essentia_test import *
+
+
+class TestTensorFlowPredictEffnetDiscogs(TestCase):
```

Comment:
Any test for what happens in the case of batch size mismatch?

Reply:
TF throws a shape mismatch exception in that case.
I don't think this should be tested because it is the responsibility of the user to know if the model requires a specific batch size or not. But we should clearly mention this in the models' page.

I designed the algorithm to be batch-size agnostic because in the future we may be interested in creating new versions of the model with different batch sizes. I even allowed the `accumulate mode` (the batch size is dynamically set to fit the whole input stream) to be consistent with other algorithms and because we may find the way to support dynamic dimensions in the future.